src/background/services/accounts/handlers/avalanche_renameAccount.ts

```diff
@@ -0,0 +1,113 @@
+import { injectable } from 'tsyringe';
+import { ethErrors } from 'eth-rpc-errors';
+
+import { DAppRequestHandler } from '@src/background/connections/dAppConnection/DAppRequestHandler';
+import {
+  DAppProviderRequest,
+  JsonRpcRequestParams,
+} from '@src/background/connections/dAppConnection/models';
+import { DEFERRED_RESPONSE } from '@src/background/connections/middlewares/models';
+import { openApprovalWindow } from '@src/background/runtime/openApprovalWindow';
+
+import { isSyncDomain } from '../../network/utils/getSyncDomain';
+import { AccountsService } from '../AccountsService';
+import { Action } from '../../actions/models';
+import { Account } from '../models';
+
+type Params = [accountId: string, newName: string];
+
+@injectable()
+export class AvalancheRenameAccountHandler extends DAppRequestHandler<
+  Params,
+  null
+> {
+  methods = [DAppProviderRequest.ACCOUNT_RENAME];
+
+  constructor(private accountsService: AccountsService) {
+    super();
+  }
+
+  handleAuthenticated = async (
+    rpcCall: JsonRpcRequestParams<DAppProviderRequest, Params>
+  ) => {
+    const { request, scope } = rpcCall;
+    const [accountId, newName] = request.params;
+
+    if (!request.site?.domain) {
+      return {
+        ...request,
+        error: ethErrors.rpc.invalidRequest({
+          message: 'Missing dApp domain information',
+        }),
+      };
+    }
+
+    const account = this.accountsService.getAccountByID(accountId);
+
+    if (!account) {
+      return {
+        ...request,
+        error: ethErrors.rpc.invalidParams({
+          message: 'No account with specified ID',
+        }),
+      };
+    }
```

Comment:
Can we also validate the `newName` param? Currently, the handler accepts empty string.
Would be great if we added the same validation to our own UI as well (it also accepts empty strings)

Reply:
Sure thing :) 6a2d3dded